modules/core/arrow-generic/src/test/kotlin/arrow/generic/CoproductTest.kt

```diff
@@ -82,5 +82,37 @@ class CoproductTest : UnitSpec() {
                     { "Third" }
             ) shouldBe "Third"
         }
+
+        "cop replacement functions are equivalent" {
+            val firstValue = "String"
+            val secondValue = 100L
+            val thirdValue = none<String>()
+
+            firstValue.cop<String, Long, Option<String>>() shouldBe firstValue.first<String, Long, Option<String>>()
+            secondValue.cop<String, Long, Option<String>>() shouldBe secondValue.second<String, Long, Option<String>>()
+            thirdValue.cop<String, Long, Option<String>>() shouldBe thirdValue.third<String, Long, Option<String>>()
+        }
+
+        "types can be inferred with the cop replacement functions" {
+            fun typeInferenceTest(input: String?): Coproduct3<String, Int, Option<String>> {
```

Comment:
would we need to add a test for inference when direcly using the ADT constructors in the function returns?